engine/include/Engine/Core/Rendering/Renderer/IRenderer.hpp

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (C) 2021 Amara Sami, Dallard Thomas, Nardone William, Six Jonathan
+ * This file is subject to the LGNU license terms in the LICENSE file
+ *	found in the top-level directory of this distribution.
+ */
+
+#pragma once
+
+#include "Engine/Core/Rendering/Window/IWindow.hpp"
+#include <memory> //std::unique_ptr
+
+namespace Engine::Core::Renderering
+{
+class IRenderer
```

Comment:
Same as [previously](https://github.com/GP-Engine-team/GP_Engine/pull/13/files?file-filters%5B%5D=.c&file-filters%5B%5D=.cpp&file-filters%5B%5D=.h&file-filters%5B%5D=.hpp&file-filters%5B%5D=.inl&file-filters%5B%5D=.lib&file-filters%5B%5D=.md&file-filters%5B%5D=No+extension&file-filters%5B%5D=dotfile&hide-deleted-files=true#r566436732).

I'm not sure this is necessary, given we have choosen our backend (OpenGL 4.6 / GLSL). I wouldn't worry about that for now, we can always extend our API to any other GPU-based graphics API later. I don't think that would be too much trouble.

On a side note, I'm not sure what `initialize()` and `release()` would be exactly here.

The idea really is to keep it simple and straight-forward to me. 

Paging you again @ThomasProg for your insight on this matter.

Reply:
Same response as the previous message

Reply:
Same as previously, I think static polymorphism is enough, since we won't be using both Vulkan and OpenGL (for instance) at the same time.